src/vs/workbench/api/common/extHost.api.impl.ts

```diff
@@ -675,6 +675,9 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 				return extHostTerminalService.terminals;
 			},
 			async showTextDocument(documentOrUri: vscode.TextDocument | vscode.Uri, columnOrOptions?: vscode.ViewColumn | vscode.TextDocumentShowOptions, preserveFocus?: boolean): Promise<vscode.TextEditor> {
+				if (URI.isUri(documentOrUri) && documentOrUri.scheme === Schemas.vscodeRemote && !documentOrUri.authority) {
```

Comment:
Let's use `IExtHostApiDeprecationService` - it only prints when the extension is under development but also/always logs telemetry so that we can reach out to bad extensions directly. That paves the way for making this an error (directly inside URI)

Reply:
Nice, learned something.